packages/playground/data-liberation/src/stream-api/WP_File_Byte_Stream.php

```diff
@@ -11,6 +11,7 @@ public function __construct( $file_path, $chunk_size = 8096 ) {
 		$this->file_path  = $file_path;
 		$this->chunk_size = $chunk_size;
 		parent::__construct();
+		$this->append_eof();
```

Comment:
I don't understand anything about this processor yet and am asking questions.

Why did this need to be added?

Reply:
The processors have two states:

* Expecting more data. Some syntax errors are then treated as "oh we're actually pausing and waiting for more data".
* No more data is coming. All errors are treated as "the input is invalid".

EOF marks a transition point between the two states.

Still, I'm questioning the need for these stream classes. They add yet another concept to learn.